hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlPrunedMemberLeftTest.java

```diff
@@ -0,0 +1,149 @@
+import org.junit.runner.RunWith;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.hazelcast.jet.sql.SqlTestSupport.createMapping;
+import static com.hazelcast.jet.sql.impl.SqlEndToEndTestSupport.calculateExpectedPartitions;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class SqlPrunedMemberLeftTest extends JetTestSupport {
+    private static final int MAP_ENTRIES = 1_000;
+    private static final String MAP_NAME = "map";
+
+    private NodeEngine nodeEngine1;
+
+    private HazelcastInstance hz1;
+    private HazelcastInstance hz2;
+    private HazelcastInstance hz3;
+    private HazelcastInstance hz4;
+    private HazelcastInstance hz5;
+
+    private Map<Address, HazelcastInstance> addrToInstanceMap;
+
+    @Before
+    public void setUp() throws Exception {
+        hz1 = createHazelcastInstance(regularInstanceConfig());
+        nodeEngine1 = getNodeEngineImpl(hz1);
+        hz2 = createHazelcastInstance(regularInstanceConfig());
+        hz3 = createHazelcastInstance(regularInstanceConfig());
+        hz4 = createHazelcastInstance(regularInstanceConfig());
+        hz5 = createHazelcastInstance(regularInstanceConfig());
+
+        addrToInstanceMap = new HashMap<>();
+        addrToInstanceMap.put(hz1.getCluster().getLocalMember().getAddress(), hz1);
+        addrToInstanceMap.put(hz2.getCluster().getLocalMember().getAddress(), hz2);
+        addrToInstanceMap.put(hz3.getCluster().getLocalMember().getAddress(), hz3);
+        addrToInstanceMap.put(hz4.getCluster().getLocalMember().getAddress(), hz4);
+        addrToInstanceMap.put(hz5.getCluster().getLocalMember().getAddress(), hz5);
+
+        prepareMap();
+        createMapping(hz5, MAP_NAME, Pojo.class, Integer.class);
+    }
+
+    @Test
+    // https://hazelcast.atlassian.net/browse/HZ-2992
+    public void test() {
+        // Given
+        final int c = 2; // constant, assuming hz3 will be coordinator.
+
+        Set<Integer> expected = IntStream.range(c * 10, (c + 1) * 10).boxed().collect(Collectors.toSet());
+        String query = "SELECT this FROM " + MAP_NAME + " WHERE f0 = " + c;
+
+        var instances = findPrunedAndNonPrunedInstances(c);
+        HazelcastInstance prunedInstanceToTerminate = instances.f0();
+        HazelcastInstance chosenCoordinator = instances.f1();
+
+        int cnt = 0;
+
+        for (SqlRow row : chosenCoordinator.getSql().execute(query)) {
+            // Read one entry and shutdown chosen member.
```

Comment:
internal buffers (Inbox etc) hold more that 1 entry, so at this time all processors except RootConsumerSink may be already done, so I am not sure if this tests what it is supposed to test.

we have a `BlockingMapStore` in some tests. Maybe we can use similar approach to simulate "slow" IMap operation to inject termination request during the query.

Reply:
I chose another way : each page is up to 4096 size, so we'll just query 2 pages (5k items).

Reply:
where did you that size from?

Reply:
`SqlStatement#DEFAULT_CURSOR_BUFFER_SIZE`

![image](https://github.com/hazelcast/hazelcast/assets/10107557/a40dcfe5-2df4-4478-9eef-6cc53817275a)

Reply:
changed to reference to the constant